protocols/kad/src/protocol.rs

```diff
@@ -333,157 +332,118 @@ pub enum KadResponseMsg {
+            r#type: proto::message::MessageType::GetValue as i32,
+            cluster_level_raw: 10,
+            key: key.to_vec(),
+            .. proto::Message::default()
+        },
+        KadRequestMsg::PutValue { record } => proto::Message {
+            r#type: proto::message::MessageType::PutValue as i32,
+            record: Some(record_to_proto(record)),
+            .. proto::Message::default()
         }
     }
 }
 
 /// Converts a `KadResponseMsg` into the corresponding protobuf message for sending.
 fn resp_msg_to_proto(kad_msg: KadResponseMsg) -> proto::Message {
     match kad_msg {
-        KadResponseMsg::Pong => {
-            let mut msg = proto::Message::new();
-            msg.set_field_type(proto::Message_MessageType::PING);
-            msg
-        }
-        KadResponseMsg::FindNode { closer_peers } => {
-            let mut msg = proto::Message::new();
-            msg.set_field_type(proto::Message_MessageType::FIND_NODE);
-            msg.set_clusterLevelRaw(9);
-            for peer in closer_peers {
-                msg.mut_closerPeers().push(peer.into());
-            }
-            msg
-        }
-        KadResponseMsg::GetProviders {
-            closer_peers,
-            provider_peers,
-        } => {
-            let mut msg = proto::Message::new();
-            msg.set_field_type(proto::Message_MessageType::GET_PROVIDERS);
-            msg.set_clusterLevelRaw(9);
-            for peer in closer_peers {
-                msg.mut_closerPeers().push(peer.into());
-            }
-            for peer in provider_peers {
-                msg.mut_providerPeers().push(peer.into());
-            }
-            msg
-        }
-        KadResponseMsg::GetValue {
-            record,
-            closer_peers,
-        } => {
-            let mut msg = proto::Message::new();
-            msg.set_field_type(proto::Message_MessageType::GET_VALUE);
-            msg.set_clusterLevelRaw(9);
-            for peer in closer_peers {
-                msg.mut_closerPeers().push(peer.into());
-            }
-            if let Some(record) = record {
-                msg.set_record(record_to_proto(record));
-            }
-
-            msg
-        }
-        KadResponseMsg::PutValue {
-            key,
-            value,
-        } => {
-            let mut msg = proto::Message::new();
-            msg.set_field_type(proto::Message_MessageType::PUT_VALUE);
-            msg.set_key(key.to_vec());
-
-            let mut record = proto::Record::new();
-            record.set_key(key.to_vec());
-            record.set_value(value);
-            msg.set_record(record);
-
-            msg
+        KadResponseMsg::Pong => proto::Message {
+            r#type: proto::message::MessageType::Ping as i32,
+            .. proto::Message::default()
+        },
+        KadResponseMsg::FindNode { closer_peers } => proto::Message {
+            r#type: proto::message::MessageType::FindNode as i32,
+            cluster_level_raw: 10,
```

Comment:
This is normally 9 but this field is unused so it doesn't matter.

Reply:
Sorry for those mistakes. A bit too much copy and paste. :blush: